core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -150,7 +147,12 @@ public TimingCollector getTimingCollector() {
      */
     @Deprecated
     public SqlStatements setTimingCollector(TimingCollector timingCollector) {
-        this.timingCollector = timingCollector == null ? TimingCollector.NOP_TIMING_COLLECTOR : timingCollector;
+        this.sqlLogger = new SqlLogger() {
```

Comment:
Probably still want the null check here. `timingCollector == null ? SqlLogger.NOP : etc`